### PR TITLE
ARTEMIS-2104 Added several tests in artemis-cli//MessageSerializerTest

### PR DESCRIPTION
**Pull request by feuillemorte**
Hello!

I wrote some tests in MessageSerializerTest file, please review it.

Added new cases:
- export from anycast queue, then import to multicast topic where number of queues > 1
- export from anycast queue, then import to multicast fqqn where number of queues > 1
- export from multicast topic, then import to anycast queue (both address)
- export from anycast queue, then import to multicast topic (both address)

Thank you!

### CODE REVIEW

**Comment by clebertsuconic** [comment]
a PR is not really a good place to ask questions.. is this supposed to show a bug.. or just help you understand it?

File: artemis-cli/src/test/java/org/apache/activemq/cli/test/MessageSerializerTest.java [hunk 1]
```diff
@@ -321,6 +329,161 @@ public void exportFromFQQN() throws Exception {
...
+
+      File file = createMessageFile();
+      int noMessages = 10;
+
+      Session session = connection.createSession(false, Session.AUTO_ACKNOWLEDGE);
+      connection.start();
+
+      createBothTypeAddress(address);
+
+      Thread export = new Thread(() -> {
```
Comment (clebertsuconic) [inline artemis-cli/src/test/java/org/apache/activemq/cli/test/MessageSerializerTest.java]: Why you need to do the export on a new thread>

File: artemis-cli/src/test/java/org/apache/activemq/cli/test/MessageSerializerTest.java [hunk 2]
```diff
@@ -321,6 +329,161 @@ public void exportFromFQQN() throws Exception {
...
+
+      export.start();
+
+      List<Message> messages = new ArrayList<>(noMessages);
+      for (int i = 0; i < noMessages; i++) {
+         messages.add(session.createTextMessage(RandomUtil.randomString()));
+      }
+
+      // wait for the listener, TODO: replace sleep to waiting loop
+      TimeUnit.SECONDS.sleep(3);
```
Comment (clebertsuconic) [inline artemis-cli/src/test/java/org/apache/activemq/cli/test/MessageSerializerTest.java]: Why the wait?

File: artemis-cli/src/test/java/org/apache/activemq/cli/test/MessageSerializerTest.java [hunk 3]
```diff
@@ -321,6 +329,161 @@ public void exportFromFQQN() throws Exception {
...
+      List<Message> messages = new ArrayList<>(noMessages);
+      for (int i = 0; i < noMessages; i++) {
+         messages.add(session.createTextMessage(RandomUtil.randomString()));
+      }
+
+      sendMessages(session, getDestination(address), messages);
+
+      exportMessages(address, noMessages, file);
+
+      Thread receive = new Thread(() -> {
```
Comment (clebertsuconic) [inline artemis-cli/src/test/java/org/apache/activemq/cli/test/MessageSerializerTest.java]: You shouldn't need any threads at all on this test.. just a sequential run.

**Comment by michalxo** [comment]
Can this be merged now?

**Pull request closed without merging**